src/vs/workbench/browser/parts/editor/breadcrumbsModel.ts

```diff
@@ -53,9 +55,9 @@ export class EditorBreadcrumbsModel {
 		private readonly _uri: URI,
 		private readonly _editor: ICodeEditor | undefined,
 		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@ITextResourceConfigurationService private readonly _textResourceConfigurationService: ITextResourceConfigurationService,
```

Comment:
The "normal" ICOnfigurationService shouldn't be needed anymore and everything should be possible using the text resource based service

Reply:
`TextResourceConfigurationService` doesn't appear to be a strict superset of ConfigurationService, and unfortunately [BreadcrumbsConfig](https://github.com/microsoft/vscode/blob/14208840271c9f764ac759db4d412365e7ab31ba/src/vs/workbench/browser/parts/editor/breadcrumbs.ts#L56) uses one of the unported methods `updateValue` right [here](https://github.com/microsoft/vscode/blob/14208840271c9f764ac759db4d412365e7ab31ba/src/vs/workbench/browser/parts/editor/breadcrumbs.ts#L101) in method `BreadcrumbsConfig.updateValue` This method is, in turn, bound and used by two breadcrumb configuration options, [here](https://github.com/microsoft/vscode/blob/9d9a99d67562315c34c195865c0720abff96c721/src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts#L532) and [here](https://github.com/microsoft/vscode/blob/9d9a99d67562315c34c195865c0720abff96c721/src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts#L585).

This more directly relates to your other coment on `BreadcrumbsControl` since the references to `updateValue` are all in there, but I'm guessing we don't want to have to remove `BreadcrumbsConfig` as a dependency of either class.

What I have right now definitely isn't the ideal solution though, so I could  try to bring `TextResourceConfigurationService` in line with `ConfigurationService` either in this PR, or get another pr merged first if this issue is too tangential.

I didn't really check to see if I could just use the one config service though, so thanks for pointing it out!

Reply:
That's for the explaining - I learnt something :-). Yeah, the `TextResourceConfigService` seems to be limiting and causing less pretty code /cc @sandy081. Let's keep the PR free from cleaning that up.

Reply:
Filed https://github.com/microsoft/vscode/issues/85884 to enhance `ITextResourceConfigService `